src/main/scala/org/clulab/reach/extern/export/fries/FriesOutput.scala

```diff
@@ -321,205 +323,209 @@ class FriesOutput extends JsonOutputter {
-    s"link-$paperId-$ORGANIZATION-$RUN_ID-${passage.chunkId}-$offset-${linkIdCntr.genNextId()}"
 
-  private def mkPassageId(paperId:String, passage:FriesEntry):String =
-    s"pass-$paperId-$ORGANIZATION-$RUN_ID-${passage.chunkId}"
-
-  private def mkSentenceId(paperId:String, passage:FriesEntry, offset:Int):String =
-    s"sent-$paperId-$ORGANIZATION-$RUN_ID-${passage.chunkId}-$offset"
-
-
-  private def mkArgument(
-    name:String,
-    arg:Mention,
-    argIndex:Int,
-    entityMap: IDed,
-    eventMap:IDed,
-    argFeatures:Option[RoleWithFeatures] = None
-  ): PropMap =
-  {
-    val m = new PropMap
-    m("object-type") = "argument"
-    m("type") = prettifyLabel(name)
+  private def makeArgument (name: String,
+                            arg: Mention,
+                            argIndex: Int,
+                            parent: Mention,
+                            entityMap: IDed,
+                            eventMap: IDed,
+                            argFeatures:Option[RoleWithFeatures] = None): PropMap = {
+    val pm = new PropMap
+    pm("object-type") = "argument"
+    pm("type") = prettifyLabel(name)
     val argType = mkArgType(arg)
-    m("argument-type") = argType
-    m("index") = argIndex
-    m("text") = arg.text
+    pm("argument-type") = argType
+    pm("index") = argIndex
+    pm("text") = arg.text
 
     argType match {
-      case "complex" =>
-        // this is a complex: print the participants
-        assert(arg.isInstanceOf[RelationMention], "complex 'arg' is not an instance of RelationMention")
+
+      case "complex" =>                     // a Protein Complex: print the participants
+        if (!isRelationMention(arg))
+          throw new RuntimeException(s"Complex argument [${arg.text}] is not an instance of RelationMention")
         val participants = new PropMap
-        val complexParticipants = arg.asInstanceOf[RelationMention].arguments
-        for(key <- complexParticipants.keySet) {
-          // FIXME: resolve each participant.  Should this be done elsewhere?
-          val ms: Seq[Mention] = complexParticipants.get(key).get.map(m => m.antecedentOrElse(m))
-          for ((p, i) <- ms.zipWithIndex) {
-            assert(p.isInstanceOf[TextBoundMention], s"complex participant is not an instance of TextBoundMention: ${p}")
-            if (!entityMap.contains(p)) {
-              throw new RuntimeException(s"Complex participant [${p.text} [mods: ${p.toCorefMention.modifications.map(_.toString).mkString(" ")}}]] not in entityMap \nin event [$currEvent] \nin sentence[${p.document.sentences(p.sentence).words.mkString(" ")}]:\n" + p.json(pretty = true))
-            }
+        val complexArgs = arg.asInstanceOf[RelationMention].arguments
+        for (key <- complexArgs.keys.toList.sorted) {
+          val argSeq = complexArgs.get(key).get
+          for ((p, i) <- argSeq.zipWithIndex) {
+            if (!isTextBoundMention(p))
```

Comment:
I guess confirming that the participant matches the label `Entity` isn't enough here, huh?  I'm just wondering about something like a complex of a complex.  We're "flattening" everything now, but who knows what we'll be doing in the future.